src/dmd/statementsem.d

```diff
@@ -123,8 +124,17 @@ private Expression checkAssignmentAsCondition(Expression e)
 // Performs semantic analysis in Statement AST nodes
 extern(C++) Statement statementSemantic(Statement s, Scope* sc)
 {
+    version (callback_API)
+        if (Compiler.onStatementSemanticStart)
+            Compiler.onStatementSemanticStart(s, sc);
```

Comment:
Maybe give `onStatementSemantic(Start|End)` a no-op implementation by default to avoid these null checks?